### PR TITLE
Add a `doc` subcommand.

### PR DESCRIPTION
**Pull request by sunfishcode**
Add a `cargo component doc` subcommand, which generates API documentation for the user's component and optionally its dependencies.

### CODE REVIEW

File: src/commands/doc.rs [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+    /// Force generation of all dependency bindings.
+    #[clap(long = "generate")]
+    pub generate: bool,
+}
+
+impl DocCommand {
+    /// Executes the command.
+    pub async fn exec(self, config: &mut Config) -> Result<()> {
+        log::debug!("executing compile command");
```
Comment (peterhuene) [inline src/commands/doc.rs:121]:
```suggestion
        log::debug!("executing document command");
```

**Review by peterhuene — APPROVED** [review]
Awesome, thanks for adding this.

Just the one minor nit.

Additionally, could we add the command to the root `README`?

**Comment by peterhuene** [comment]
Also, could we add a test file for the `doc` command for, at a minimum, testing `--help`, but ideally if we could put a doc comment on the component and check that `target/doc/<name>/index.html` contains its content?

**Review by peterhuene — APPROVED** [review]

**Comment by sunfishcode** [comment]
It turns out I added the test wrong, so it's passing for the wrong reason. Fixing this will depend in https://github.com/bytecodealliance/wasm-tools/pull/960 to actually propagate the documentation comment to the bindings, so that it shows up in the generated documentation.
